test/system/helpers.systemd.bash

```diff
@@ -14,12 +14,44 @@ fi
 
 mkdir -p $UNIT_DIR
 
+journalctl_support_user() {
+    r_value=0
+    run systemd-analyze cat-config systemd/journald.conf | grep Storage
+    if [[ $status != 0 ]]; then
+        echo "Can not get journald.conf, don't use --user in following tests."
+    fi
+
+    if [[ "$output" == "Storage=persistent" ]]; then
+        r_value=1
+    fi
```

Comment:
If you actually have a drop in file cat-config prints all file outputs while this only catches one, so I am not sure this is is even correct.

I would suggest you do a proper function test instead by checking if journalctl --user returns logs and laso the result should be cached so we don't always run this command over and over again.

Lastly isn't this just inverse of what you are saying, 0 return should mean success but you return 1 instead which is super confusing for any caller.